graph/src/main/scala/net/spals/dsa/graph/Graph.scala

```diff
@@ -30,18 +30,18 @@ case class WeightedGraph[T, W](vertices: Set[Vertex[T]], edgeFunc: (Vertex[T], V
     WeightedGraph(vertices + v, edgeFunc)
   }
 
-  def getEdge(v1: Vertex[T], v2: Vertex[T]): Option[W] = {
-    edgeFunc.apply(v1, v2)
+  def getEdge(v1: Vertex[T], v2: Vertex[T]): Option[Edge[T, W]] = {
+    edgeFunc.apply(v1, v2).map(Edge(v1, v2, _))
```

Comment:
_ is syntactic sugar which refers to the input of the function. For example, the following are equivalent:

```scala
Option("   Str   ").map(s => s.trim())
```
```scala
Option("   Str   ").map(_.trim)
```